### ISSUE TITLE
yay -Qu / checkupdates differences / check for updates without sync

### ISSUE THREAD

**Issue by squalou** [Jguer/yay#1350]
**Is your feature request related to a problem? Please describe.**

I wish I could get a single command for yay to display available updates on repos and aur, without any DB update.
An extended 'checkupdates' if you like. Without 'sudo' auth, for easy scripting.

Right now for instance, if I run `checkupdates` I'll get 3 packages

```
mesa 20.1.5-1 -> 20.1.6-1
mesa-vdpau 20.1.5-1 -> 20.1.6-1
vulkan-intel 20.1.5-1 -> 20.1.6-1
```

but yay -Qu says nothing.  (neither does pacman -Qu)

So, a "Sync" is missing somewhere but I'm a bit lost. Apparently checkupdates works on a local copy of main DB, so it does not perform a sync for real.

If I perform yay -Sy, I'm asked for auth (which is fine), then yay -Qu output is fine, *but* I did update main DB and had to enter password. Unsuitable for a basic scripting.

Finally, if there are updates on AUR, yay -Qu will tell, even without the -Sy part.

As a result ... it's a bit hard to follow.

**Describe the solution you'd like**

`yay -Qu --some-option`  to check repo like checkupdate does, without db update, without sudo password.

**Describe alternatives you've considered**

Right now, I get output of checkupdates, then get the output of yay -Qu, and merge the two lists.
(all this in bash)
Have to eliminate duplicates. 

I did try to replace 'pacman' by 'yay' in regular 'checkupdates' but then it asks for sudo password, which I want to avoid.

currently I have this :
```
#!/bin/bash
CHK=`checkupdates`
CHKAUR="$(yay -Qu)"

if [ ${#CHKAUR} -gt 0 ]; then 
    if [ "$CHK" = "" ]; then
        CHK="${CHKAUR}"
    else
        CHK="${CHK}"$'\n'"${CHKAUR}"
    fi
fi

CHK=$(echo "${CHK}" | sort |uniq)
echo "${CHK}"
```

not a big deal, could be nicer and shorter probably

**Comment by Jguer** [comment]
I see how it was implemented, seems pretty doable, I'll add it as an option

**Comment by Morganamilo** [comment]
I've always advocated to do `checkupdates && yay -Qua`.

**Comment by squalou** [comment]
indeed ! good idea. Thanks.

**Comment by blaxpot** [comment]
I'd also love to see something like `checkupdates` for the AUR. The workarounds suggested here are helpful and all, but are missing the features allowed by having a separate, temporary package DB (this is how `checkupdates` works).

For example, I like to have a counter in my prompt telling me how many updates are available. Since actually going out and checking for updates takes a while, I don't want to do that every time a prompt appears, as that would mean waiting a few seconds every time. Luckily with a temp package DB I can just check for updates every n minutes using a `cron` job and use `checkupdates --nosync` in my prompt to avoid syncing the package DB.

It would also be nice to download updates to a cache before installing them, which can be done with `checkupdates --download`.

AFAIK there's no tool that does a similar thing for AUR packages, though there are several that list available AUR updates (e.g. `yay -Qua`/`packaur -k`/`pacman -Qm | aur vercmp`).

Come to think of, though, maybe it makes more sense to have this functionality in a separate tool rather than in an AUR helper, as is the case with `checkupdates` itself.
